### DISCUSSION TITLE
halting the creation process

### DISCUSSION THREAD

**Discussion by zbadshah** [filamentphp/filament#4734] [Showcase]
I am using $this->halt as suggested in the hook within create page but getting this error.
App\Filament\Resources\SalariesResource\Pages\CreateSalaries::halt does not exist.

Please let me know if I am missing something.

**Comment by danharrin** [comment]
You're probably not on the latest version

**Reply by zbadshah** [reply]
Thank you for your response Dan.
I am on version 2 and also used upgrade command
What else can I do to ensure that I am on latest version?

**Reply by danharrin** [reply]
`composer update` and make sure there arent restrictions in composer.json. Check your version with `php artisan about`

**Reply by zbadshah** [reply]
Thanks a lot. It appears to be working now.

**Reply by zbadshah** [reply]
Hi Dan,
1. Using Filament v2.16.58. Great work. 
Failed $this->halt in afterFill() hook function.
I tried updating etc.
Kindly suggest if there is something I can do to avoid this error.

2. Please provide help under tricks page to access only the table rows that belong to the logged in user. I know you did cover it in one of the videos, but it is not working somehow. Thanks

**Reply by zbadshah** [reply]
BTW it is working with other hooks.

**Reply by delavago** [reply]
> Hi Dan,
> 
>     1. Using Filament v2.16.58. Great work.
>        Failed $this->halt in afterFill() hook function.
>        I tried updating etc.
>        Kindly suggest if there is something I can do to avoid this error.
> 
>     2. Please provide help under tricks page to access only the table rows that belong to the logged in user. I know you did cover it in one of the videos, but it is not working somehow. Thanks

I am so late to this but, halt is normally used to stop/halt the creation process. the after fill hook is not apart of that process
